src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3860,6 +3781,90 @@ private void TrackNullableStateOfNullableValue(BoundExpression node, BoundExpres
             }
         }
 
+        private void TrackNullableStateOfTupleConversion(BoundConversion node)
+        {
+            Debug.Assert(node.ConversionKind == ConversionKind.ImplicitTuple || node.ConversionKind == ConversionKind.ExplicitTuple);
+
+            int valueSlot = MakeSlot(node.Operand);
+            if (valueSlot < 0)
+            {
+                return;
+            }
+
+            int slot = GetOrCreateObjectCreationPlaceholderSlot(node);
+            if (slot < 0)
+            {
+                return;
+            }
+
+            TrackNullableStateOfTupleConversion(node.Conversion, node.Type, node.Operand.Type, slot, valueSlot);
+        }
+
+        private void TrackNullableStateOfTupleConversion(Conversion conversion, TypeSymbol targetType, TypeSymbol operandType, int slot, int valueSlot)
+        {
+            Debug.Assert(conversion.Kind == ConversionKind.ImplicitTuple || conversion.Kind == ConversionKind.ExplicitTuple);
+            Debug.Assert(slot > 0);
+            Debug.Assert(valueSlot > 0);
+
+            var valueTuple = operandType as TupleTypeSymbol;
+            if (valueTuple is null)
```

Comment:
Consider using pattern matching `if (operandType is TupleTypeSymbol valueTuple)`